Systems/c172p-engine.xml

```diff
@@ -79,6 +79,60 @@
             </clipto>
             <output>/engines/active-engine/rpm-over-redline</output>
         </fcs_function>
+
+        <fcs_function name="damage level calculation">
+              <function>
+                <ifthen>
+                    <eq>
+                        <p>/engines/active-engine/damage_allowed</p>
+                        <v>1</v>
+                    </eq>
+                    <sum>
+                        <property>/engines/active-engine/damage-level</property>
+                        <product>
+                           <table>
+                                <independentVar lookup="row">/engines/active-engine/rpm-over-redline</independentVar>
+                                <tableData>
+                                    0    0
+                                  100    1
+                                  200    4
+                                  300    9
+                                  400   16
+                                  500   25
+                                  600   36
+                                  700   49
+                                  800   64
+                                  900   81
+                                 1000  100
+                                 1100  121
+                                 1200  144
+                                 1300  169
+                                 1400  196
+                                 1500  225
+                                 9999  300
+                                </tableData>
+                            </table>
+                            <property>simulation/channel-dt</property>
+                        </product>
+                    </sum>
+                    <v>0</v> <!-- this is the value if damage is disabled -->
+                </ifthen>
+            </function>
+            <output>/engines/active-engine/damage-level</output>
+        </fcs_function>
+		
+		<fcs_function>
```

Comment:
Tabs! :cry: In case you press the actual tab key for indentation: editors provide the ability to replace those automatically with a number of spaces. See the preferences/settings of your editor.

Reply:
Yes, I thought I had done that for each changed file, but obviously made changes afterwards! Old habits are hard to break ;)